addons/resource/tests/test_resource.py

```diff
@@ -584,3 +584,67 @@ def test_work_days_count_timezones_ultra(self):
             to_naive_utc(Datetime.from_string('2013-02-12 20:00:00'), self.env.user))
         self.assertEqual(res['days'], 0.75)
         self.assertEqual(res['hours'], 6.0)
+
+class TestGlobalLeaves(TestResourceCommon):
+
+    def setUp(self):
+        super(TestGlobalLeaves, self).setUp()
+
+        self.Leave = self.env['resource.calendar.leaves']
+
+        self.global_no_tz = self.Leave.create({
+            'name': 'global no timezone',
+            'calendar_id': self.calendar.id,
+            'date_from': to_naive_utc(Datetime.from_string('2018-12-25 08:00:00'), self.env.user),
+            'date_to': to_naive_utc(Datetime.from_string('2018-12-25 23:00:00'), self.env.user),
+            'tz': False,
+        })
+
+        self.global_user_tz = self.Leave.create({
+            'name': 'global user timezone',
+            'calendar_id': self.calendar.id,
+            'date_from': to_naive_utc(Datetime.from_string('2018-05-01 08:00:00'), self.env.user),
+            'date_to': to_naive_utc(Datetime.from_string('2018-05-01 23:00:00'), self.env.user),
+        })
+
+        # Leave from 8:00 to 13:00 but we have to adapt cause we define it in UTC
+        self.global_spec_tz = self.Leave.create({
+            'name': 'global specific timezone',
+            'calendar_id': self.calendar.id,
+            'date_from': Datetime.from_string('2018-04-05 23:00:00'),
+            'date_to': Datetime.from_string('2018-04-06 04:00:00'),
+            'tz' : 'Japan', # UTC+9
+        })
+
+    def test_timezones(self):
+        self.assertEqual(self.global_no_tz.tz, False)
+        self.assertEqual(self.global_user_tz.tz, self.env.user.tz)
+        self.assertEqual(self.global_spec_tz.tz, 'Japan')
+
+    def test_leave(self):
+        def to_tuple(interval):
+            return (interval.start_datetime, interval.end_datetime)
+
+        work_no_tz = [to_tuple(x) for x in self.calendar._get_day_leave_intervals(Date.from_string('2018-12-25'), time(0), time(23, 59, 59), self.resource1_id)]
+        self.assertEqual(work_no_tz[0], (Datetime.from_string('2018-12-25 08:00:00'), Datetime.from_string('2018-12-25 16:00:00')))
+
+        work_user_tz = [to_tuple(x) for x in self.calendar._get_day_leave_intervals(Date.from_string('2018-05-01'), time(0), time(23, 59, 59), self.resource1_id)]
+        self.assertEqual(work_user_tz[0], (Datetime.from_string('2018-05-01 08:00:00'), Datetime.from_string('2018-05-01 16:00:00')))
+
+        work_spec = [to_tuple(x) for x in self.calendar._get_day_leave_intervals(Date.from_string('2018-04-06'), time(0), time(23, 59, 59), self.resource1_id)]
```

Comment:
Tests are not easy to read. I think some comments could help. For example here if I understand well you take into account the japen-tz leaves. I think a small comment telling what should happen would be welcomed. Otherwise I fear someone breaking the tests will simply update the values without really digging into the code ^^